### ISSUE TITLE
API Documentation

### ISSUE THREAD

**Issue by Smiley01987** [VisualDataWeb/WebVOWL#90]
Hi,

I would like to use the WebVOWL library in my project, but i cant seem to find any documentation on how to implement this?

Cheers

**Comment by steffen-l** [comment]
What information do you miss? Could you detail your request?

**Comment by Smiley01987** [comment]
Well i would like some documentation on how to implement the library in my current project.

**Comment by steffen-l** [comment]
See http://vowl.visualdataweb.org/webvowl.html - especially the "linking" and installation sections. Does this info help?

**Comment by Smiley01987** [comment]
Not really. The information stated in those sections are already fully build web applications. What i need is some documentation on how to exactly implement the webvowl.js library.  Just the visualisation.

**Comment by linkvt** [comment]
I think @Smiley01987 asks for some kind of API documentation for the WebVOWL code to implement his own app.

Since there is no npm package or something similar for WebVOWL, you would need to include the built webvowl.js on your own in your project. 
The best "documentation" available is the code you can find in the WebVOWL app, e.g. here: [app.js](https://github.com/VisualDataWeb/WebVOWL/blob/master/src/app/js/app.js). You can see how to setup the graph, use modules/plugins and control the graph. It's currently afaik the best way to understand how to use the webvowl.js code.

**Comment by steffen-l** [comment]
Thanks for clarifying, Vincent. We will consider to improve the separation of the actual VOWL visualization and the GUI / controls in the future. npm would be one possible option, right. For now, you can only follow the approach sketched by Vincent above. The example from the VOWL 2 spec (http://vowl.visualdataweb.org/v2/#example) may also help (though note it is oudated).

**Comment by nilsgumpfer** [comment]
Hi Steffen, 
could you maybe refine a little bit what your future plans are about (Web)VOWL?
Best Regards, Nils

**Comment by steffen-l** [comment]
We are currently working on improving the orientation, exploration, and navigation in larger ontology graphs. After adding an improved search result navigation and smooth zoom in the last minor releases, the next major release will likely include some sophisticated semantic zooming and local exploration functionalities - however, an effective and convincing solution requires quite some research as well as user evaluations and testing, so it might take a while until we can finally release it to the public. Apart from that, I can imagine a lot of useful and cool features and extensions (ABox visualization, interactive editing, etc.), but these are major development tasks and it is hard to give any concrete development plan for them.

**Comment by nilsgumpfer** [comment]
Sounds really nice! ..do you have a rough timeline, maybe also if one is able to get a first alpha version or something for testing and evaluation? :) Thx

**Comment by steffen-l** [comment]
With regard to the semantic zooming and local exploration we are currently working on, it should be safe to say that there will likely be a release in June or July. Hope this helps.

**Comment by stormbel** [comment]
Hello!
I have a little problem: I need to use the WebVOWL library in my project, but I can't find API documentation for the WebVOWL code to implement his own app.

Can you help me, please?

**Comment by kipavy** [comment]
Hi, still no doc 9 years later ?
